clusterloader2/testing/measurements-gatherer/config.yaml

```diff
@@ -0,0 +1,147 @@
+#Constants
+{{$LARGE_BACKEND_SIZE := 300}}
+{{$MEDIUM_BACKEND_SIZE := 150}}
+{{$SMALL_BACKEND_SIZE := 10}}
+{{$LARGE_BACKEND_LABEL := "ilb-large"}}
+{{$MEDIUM_BACKEND_LABEL := "ilb-medium"}}
+{{$SMALL_BACKEND_LABEL := "ilb-small"}}
+# adding a fixed value for first version of the test, rate of pod creation not a concern yet.
+{{$ilbQPS := 500}}
+{{$namespaces := 1}}
+
+#Test
+name: ilbload
+automanagedNamespaces: {{$namespaces}}
+tuningSets:
+- name: ILBConstantQPS
+  qpsLoad:
+    averageQps: {{$ilbQPS}}
+steps:
+# Measure each of the ILB services separately, this will provide insight on how long programming
+# ILB takes as a function of number of backends.
+- measurements:
+  - Identifier: CreateServices
```

Comment:
This is really great!

But I'm thinking now whether we shouldn't take it a step further :)

The only thing that I don't like here is that we have this meta-measurement for executing multiple measurements. 
You get used to it, but it's not very beautiful and there is still some boilerplate (e.g. extra Identifier, method: GatherMeasurements, nested Params, etc.)
I think that it would be cleaner if it was part of the [ClusterLoader2 Measurement API](https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/api/types.go#L147).

We could do the following, change the `Identifier` type to interface{}, document that identifier should be either string or map (and enforce that in the code, e.g. throw error if it's not). If Identifier is a string, it works as a regular Measurement. If it's a map (from string to "params") then it works as your "GatherMeasurements", e.g. in that case the json would be

```
steps:
- measurements:
  - Method: ServiceCreationLatency
    Identifier:
      ServiceCreationLatencyLarge:
       labelSelector: size = {{$LARGE_BACKEND_LABEL}}
      ServiceCreationLatencyMedium:
       labelSelector: size = {{$MEDIUM_BACKEND_LABEL}}
      ServiceCreationLatencySmall:
       labelSelector: size = {{$SMALL_BACKEND_LABEL}}
    Params:
      action: start
  - Identifier: WaitForRunningDeployments
    Method: WaitForControlledPodsRunning
    Params:
      action: start
      apiVersion: apps/v1
      kind: Deployment
      labelSelector: group = ilb-load
      operationTimeout: 15m
```

WDYT?

You could reuse the code you already wrote for that, instead of creating a new measurement, we'd create a new util, e.g. measurement_executor, that would encapsulate all this logic (validation, executing "single" measurement, executing "multi-measurement"), and we'd call it from simple_test_executor.go. WDYT?

Reply:
Well, what you are saying is correct :) I will look at the approach you posted. My gut feel is, the approach you posted can also be re-used in this other issue regarding SLIs:

https://github.com/kubernetes/perf-tests/issues/787#issuecomment-593316632.

Reply:
@mm4tt , I implemented these changes. Turns out, the code and the configs have become a lot cleaner now :) However, some of the changes I made from earlier works because of the decision we had taken then needed to be reverted(like adding measurementmanager to measurementconfig).

Let me know, whichever ones you need me to send as part of a separate PR. Mainly because changing the identifier type to interface{} has forced changes on a couple of measurements.